src/main/java/com/watch/shop/app/controller/WatchController.java

```diff
@@ -0,0 +1,57 @@
+package com.watch.shop.app.controller;
+
+import com.watch.shop.app.model.entity.Watch;
+import com.watch.shop.app.model.service.WatchService;
+import com.watch.shop.app.view.WatchView;
+import lombok.RequiredArgsConstructor;
+
+import static com.watch.shop.app.util.Constants.INVALID_CHOICE_MESSAGE;
+import static com.watch.shop.app.util.Constants.INVALID_INPUT_MESSAGE;
+
+@RequiredArgsConstructor
+public class WatchController {
+
+    private final WatchService watchService;
+    private final WatchView watchView;
+    private final MenuHandler menuHandler;
+
+    public void run() {
+        boolean running = true;
+
+        while (running) {
+            String choice = menuHandler.printMainMenu();
+
+            switch (choice) {
+                case "1" -> menuHandler.displayAllWatches(watchService.getWatches());
+                case "2" -> sortChoice();
+                case "3" -> menuHandler.displayTotalCost(watchService.getTotalCost());
+                case "4" -> {
+                    try {
+                        Watch watch = menuHandler.printNewWatchMenu();
+                        watchService.insertWatch(watch);
+                    } catch (Exception e) {
+                        watchView.printMessage(INVALID_INPUT_MESSAGE);
+                    }
+                }
+                case "5" -> running = false;
+                default -> watchView.printMessage(INVALID_CHOICE_MESSAGE);
+            }
+        }
+    }
+
+    private void sortChoice() {
+        boolean running = true;
```

Comment:
please use Java naming convention for boolean variables
in this case, isExit for example:
while (!isExit) {
...
 4 - isExit = true
 
 or 
 
 while(isRunning)
 4. isRunning = false;